homeassistant/components/vilfo/sensor.py

```diff
@@ -0,0 +1,91 @@
+"""Support for Vilfo Router sensors."""
+from homeassistant.helpers.entity import Entity
+
+from .const import (
+    ATTR_API_DATA_FIELD,
+    ATTR_DEVICE_CLASS,
+    ATTR_ICON,
+    ATTR_LABEL,
+    ATTR_UNIT,
+    DOMAIN,
+    ROUTER_DEFAULT_MODEL,
+    ROUTER_DEFAULT_NAME,
+    ROUTER_MANUFACTURER,
+    SENSOR_TYPES,
+)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add Vilfo Router entities from a config_entry."""
+    vilfo = hass.data[DOMAIN][config_entry.entry_id]
+
+    sensors = []
+
+    device_info = {
+        "identifiers": {(DOMAIN, vilfo.host, vilfo.mac_address)},
+        "name": ROUTER_DEFAULT_NAME,
+        "manufacturer": ROUTER_MANUFACTURER,
+        "model": ROUTER_DEFAULT_MODEL,
+        "sw_version": vilfo.firmware_version,
+    }
+
+    for sensor_type in SENSOR_TYPES:
+        sensors.append(VilfoRouterSensor(sensor_type, vilfo, device_info))
+
+    async_add_entities(sensors, True)
+
+
+class VilfoRouterSensor(Entity):
+    """Define a Vilfo Router Sensor."""
+
+    def __init__(self, sensor_type, api, device_info):
+        """Initialize."""
+        self.api = api
+        self.sensor_type = sensor_type
+        self._device_info = device_info
+        self._unique_id = f"{self.api.unique_id}_{self.sensor_type}"
+        self._state = None
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return self._device_info
+
+    @property
+    def device_class(self):
+        """Return the device class."""
+        return SENSOR_TYPES[self.sensor_type].get(ATTR_DEVICE_CLASS)
+
+    @property
+    def icon(self):
+        """Return the icon for the sensor."""
+        return SENSOR_TYPES[self.sensor_type][ATTR_ICON]
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        parent_device_name = self._device_info["name"]
+        sensor_name = SENSOR_TYPES[self.sensor_type][ATTR_LABEL]
+        return f"{parent_device_name} {sensor_name}"
```

Comment:
`parent_device_name` will always be the same (`ROUTER_DEFAULT_NAME`), not that unique ?

Did I miss something ?

Reply:
If the user manually renames the parent device (the router in this case), the name will propagate to the child entities (if you haven't customized their respective names). See attached screenshots.

### Before renaming

![before_renaming](https://user-images.githubusercontent.com/659680/74342953-981c6c00-4daa-11ea-9967-6217460f2efb.png)

### After renaming

![after_renaming](https://user-images.githubusercontent.com/659680/74342956-98b50280-4daa-11ea-8434-06910217d36a.png)

Reply:
Ok then, I missed something 😅

Resolving